addons/mail/static/src/core/common/suggestion_service.js

```diff
@@ -163,15 +156,24 @@ export class SuggestionService {
                 cleanTerm(partner.name).includes(cleanedSearchTerm) ||
                 (partner.email && cleanTerm(partner.email).includes(cleanedSearchTerm))
             ) {
-                suggestions.push(partner);
+                suggestions.push(this.store.Mention.new({ partner }));
             }
         }
-        return {
-            type: "Partner",
-            suggestions: sort
-                ? [...this.sortPartnerSuggestions(suggestions, cleanedSearchTerm, thread)]
-                : suggestions,
-        };
+        suggestions.push(
+            ...Object.values(this.store.Mention.specialMentions)
+                .filter(
+                    (special) =>
+                        thread &&
```

Comment:
thread should not be checked here as well. should be checked outside of the filter